### PR TITLE
Remove GaussianAbsorption1D model from functional models

### PR DESCRIPTION
**Pull request by pllim**
Part of #5774

### CODE REVIEW

File: astropy/modeling/tests/test_models_quantities.py [hunk 1]
```diff
@@ -27,9 +27,6 @@
 except ImportError:
     HAS_SCIPY = False
 
-# TODO: GaussianAbsorption1D doesn't work with units because the 1- part doesn't
-# have units. How do we want to deal with that?
```
Comment (pllim) [inline astropy/modeling/tests/test_models_quantities.py]: @astrofrog , problem solved! 😂

**Comment by pllim** [comment]
Failure is from `astropy/tests/tests/test_skip_remote_data.py` that has nothing to do with this PR. This same failure has also been reported by @MSeifert04 in another PR.

**Comment by MSeifert04** [comment]
That one is extremely flaky... Ether `BadGateWay` or `TimeOut`... It helps restarting the test several times. Speaking from experience it will work after ~3 retries every 1-2 hours.

**Comment by mhvk** [comment]
Raise an issue about that test - if it is that painful, we probably should disable it...

**Pull request merged**